platforms/core-configuration/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/DefaultPrecompiledScriptPluginsSupport.kt

```diff
@@ -565,8 +565,8 @@ fun Project.declareScriptPlugins(scriptPlugins: List<PrecompiledScriptPlugin>) {
     gradlePlugin.apply {
         for (scriptPlugin in scriptPlugins) {
             plugins.create(scriptPlugin.id) {
-                it.id = scriptPlugin.id
```

Comment:
❓ 

So doesn't  `import org.gradle.kotlin.dsl.assign` help here?

Reply:
It doesn't work in this project, since we don't apply compiler plugin with `=` support